### PR TITLE
feat: adding rooms and tracks section to session page

### LINKED ISSUE #4807: Scheduler: Implement schedule page and subviews tracks and rooms in list view
The website generator offers a schedule view with the subpages "tracks" and "rooms", e.g. https://sched.eventyay.com/FOSSASIASummit/schedule.html. Please implement something similar as sorting options/views as follows:

* [x] Schedule view is same as now here e.g. https://eventyay.com/e/8fa7fd14/schedule **_DONE ALREADY_**
* [ ] Tracks view - all sessions are sorted in list view according to date, time and track -> There needs to be a sorting option according to "Tracks"
* [ ] Rooms view - all sessions are sorted in list view according to date, time and room -> There needs to be a sorting option "Rooms"

Screenshot from 2020-11-25 12-19-44

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #4807

#### Short description of what this resolves:
Currently making all changes to session page for review and decision that where these pages would be implemented.
Screenshot from 2020-08-26 23-40-27
Screenshot from 2020-08-26 23-40-23
Screenshot from 2020-08-26 23-40-11

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
So, we had a few discussions in the last meeting, but I am not sure, if we are all have the same understanding. The issue description might be confusing as well. So, lets try to decide here.

I think the page should have a row on the top with three options:
1. Schedule - shows the current schedule calendar view
2. Tracks - shows all tracks and sessions below (sessions below the track name headings are sorted by date/time)
3. Rooms - shows all rooms and sessions below (sessions below the room names headings are sorted by date/time)

Does it make sense?

**Comment by mariobehling** [comment]
This would be similar to the website generated site, but all on one page.

Screenshot from 2020-08-26 20-45-45

Compare for example to the tracks page here http://2019.fossasia.org/event/tracks.html

Then there is the schedule and rooms option.

**Comment by mariobehling** [comment]
The main difference is how sessions are sorted:
* in calendar view
* in list view clustered by tracks and sorted by time
* in list view clustered by rooms and sorted by time

**Comment by iamareebjamal** [comment]
@maze-runnar Call me in the morning if you still have some issues

File: app/controllers/public/schedule.js [hunk 1]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
   get isSchedulePublished() {
     return this.model.event.schedulePublishedOn.toISOString() !== moment(0).toISOString();
   }
+
+   hideSchedule = true;
+   hideTrack = false;
+   hideRooms = false;
+  @action
+  showSchedule() {
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:17]: Codacy found an issue: [Expected indentation of 3 spaces but found 2.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057&bid=19895647)

File: app/controllers/public/schedule.js [hunk 2]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
+    this.toggleProperty('hideSchedule');
+    this.toggleProperty('hideTrack');
+    this.toggleProperty('hideRooms');
+
+	  this.set('hideSchedule', true);
+	  this.set('hideTrack', false);
+	  this.set('hideRooms', false);
+
+  }
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:26]: Codacy found an issue: [Expected indentation of 3 spaces but found 2.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 3]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
   }
+
+   hideSchedule = true;
+   hideTrack = false;
+   hideRooms = false;
+  @action
+  showSchedule() {
+    this.toggleProperty('hideSchedule');
+    this.toggleProperty('hideTrack');
+    this.toggleProperty('hideRooms');
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:20]: Codacy found an issue: [Expected indentation of 5 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 4]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
+
+	  this.set('hideSchedule', true);
+	  this.set('hideTrack', false);
+	  this.set('hideRooms', false);
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:24]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 5]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
   get isSchedulePublished() {
     return this.model.event.schedulePublishedOn.toISOString() !== moment(0).toISOString();
   }
+
+   hideSchedule = true;
+   hideTrack = false;
+   hideRooms = false;
+  @action
+  showSchedule() {
+    this.toggleProperty('hideSchedule');
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:18]: Codacy found an issue: [Expected indentation of 5 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 6]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
+    this.toggleProperty('hideTrack');
+    this.toggleProperty('hideRooms');
+
+	  this.set('hideSchedule', true);
+	  this.set('hideTrack', false);
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:23]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 7]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
+   hideSchedule = true;
+   hideTrack = false;
+   hideRooms = false;
+  @action
+  showSchedule() {
+    this.toggleProperty('hideSchedule');
+    this.toggleProperty('hideTrack');
+    this.toggleProperty('hideRooms');
+
+	  this.set('hideSchedule', true);
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:22]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 8]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
+	  this.set('hideRooms', false);
+
+  }
+
+  @action
+  showTrack() {
+    this.toggleProperty('hideSchedule');
+    this.toggleProperty('hideTrack');
+    this.toggleProperty('hideRooms');
+
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:33]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js [hunk 9]
```diff
@@ -9,4 +9,43 @@ export default class ScheduleController extends Controller {
...
     return this.model.event.schedulePublishedOn.toISOString() !== moment(0).toISOString();
   }
+
+   hideSchedule = true;
+   hideTrack = false;
+   hideRooms = false;
+  @action
+  showSchedule() {
+    this.toggleProperty('hideSchedule');
+    this.toggleProperty('hideTrack');
```
Comment (niranjan94) [inline app/controllers/public/schedule.js:19]: Codacy found an issue: [Expected indentation of 5 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057&bid=19895647)

File: app/controllers/public/schedule.js (hunk 7 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:22]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 6 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:23]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 5 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:18]: Codacy found an issue: [Expected indentation of 5 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 8 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:33]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 9 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:19]: Codacy found an issue: [Expected indentation of 5 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 1 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:17]: Codacy found an issue: [Expected indentation of 3 spaces but found 2.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 3 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:20]: Codacy found an issue: [Expected indentation of 5 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 4 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:24]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

File: app/controllers/public/schedule.js (hunk 2 above)
Comment (niranjan94) [inline app/controllers/public/schedule.js:26]: Codacy found an issue: [Expected indentation of 3 spaces but found 2.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077057)

**Comment by maze-runnar** [comment]
Screenshot from 2020-08-27 10-20-37
Screenshot from 2020-08-27 10-20-34
Screenshot from 2020-08-27 10-20-25

**Comment by iamareebjamal** [comment]
@maze-runnar This is quite an elegant implementation in terms of code. But unfortunately, it won't work if there are more than 20 sessions in a track or room. I wish this could work. 

Also, the filters won't be applied to the tracks and room section

**Comment by maze-runnar** [comment]
> it won't work if there are more than 20 sessions in a track or room.
What should I do for this?

**Comment by iamareebjamal** [comment]
1. Load all sessions in one go and then group by in frontend. This is very inefficient and will slow down the page and server a lot.
2. You have to use sort by in API. Sort by track or sort by microlocation will group the sessions accordingly

**Comment by maze-runnar** [comment]
> You have to use sort by in API. Sort by track or sort by microlocation will group the sessions accordingly

can it be implemented same as sort by title and name in drop down like sessions?

**Comment by iamareebjamal** [comment]
Yes, but UI has to be separate

**Comment by maze-runnar** [comment]
```
const sessions = await event.query('sessions', {
      include      : 'speakers,microlocation,track',
      filter       : scheduledFilterOptions,
      sort         : '',
      'page[size]' : 0
    });
```

what would be pass in sort as param, `track.name` is not working

**Comment by iamareebjamal** [comment]
Checked that server does not support nested sort query. Please work on other issues, I'll see what needs to be done in this case

**Comment by maze-runnar** [comment]
> Checked that server does not support nested sort query. Please work on other issues, I'll see what needs to be done in this case

on which issue I should work next?

**Comment by iamareebjamal** [comment]
Linking session to session page and then on Forms and Custom Forms milestone

**Comment by maze-runnar** [comment]
> Linking session to session page and then on Forms and Custom Forms milestone

ok

> Linking session to session page and then on Forms and Custom Forms milestone

ok

**Comment by iamareebjamal** [comment]
Requirement changed, so outdated

**Pull request closed without merging**